warehouse/README.md

```diff
@@ -3,9 +3,60 @@
+1. Follow the installation instructions at https://cloud.google.com/sdk/docs/install
+   1. If this is your first time using the terminal, we recommend reading https://blog.teamtreehouse.com/introduction-to-the-mac-os-x-command-line
+      or another tutorial first. You will generally need to understand `cd` and
+      the concept of the "home directory" aka `~`. When you first open the
+      terminal, your "working directory" will be the home directory. Running the
+      `cd` command without any arguments will set your working directory back to
+      `~`.
+   2. Use `tar -xvf <drag file from Finder window to get file name> ~` to unzip into your home directory
+   3. Answer `yes` to adding the tool to your path
+   4. Most recent macOS versions use `zsh` as the default shell, so ensure the path modification is added to `~/.zshrc` when prompted
+   5. Restart your terminal, and run `gcloud init`
+   6. Answer `yes` to log in, and select the Google account associated with GCP
+   7. Set `cal-itp-data-infra` as the default project, and do not set a region
+3. You should also [set the application default](https://cloud.google.com/sdk/gcloud/reference/auth/application-default)
+4. If `bq ls` shows output, you are good to go.
+
+
+### Install poetry and Python/dbt dependencies
+1. Install [poetry](https://python-poetry.org/docs/#osx--linux--bashonwindows-install-instructions) (used for package/dependency management).
+   1. If `curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py | python -`
+      does not work, you can `curl` to a file `curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py`
+      and then execute the file with `python get-poetry.py`.
+   2. Answer `yes` to adding the tool to your path
+2. Restart your terminal and confirm `poetry --version` works.
+3. `poetry install` to create a virtual environment and install requirements
+   1. Now would be the time to [clone](https://docs.github.com/en/repositories/creating-and-managing-repositories/cloning-a-repository)
+      the `data-infra` repo if you haven't already. Use [SSH](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/adding-a-new-ssh-key-to-your-github-account),
+      not HTTPS.
+   2. This needs to be run from within the `warehouse/` directory; you may need
+      to `cd` to it via `cd <git-repos-path>/data-infra/warehouse/` or similar.
+   3. If this doesn’t work because of an error with Python version, you may need to install Python 3.9
+   4. `brew install python@3.9`
+   5. `brew link python@3.9`
+   6. After restarting the terminal, confirm with `python3 --version` and retry `poetry install`
+4. `poetry run dbt deps` to install the dbt dependencies defined in `packages.yml` (such as `dbt_utils`)
+
+### Initialize your dbt profiles.yml
+1. `poetry run dbt init` inside `warehouse/` will create a `.dbt/` directory in your home directory and a `.dbt/profiles.yml` file.
```

Comment:
I still think we need specific instructions on how to find+edit a hidden file on macos.

Maybe something like:

"
Note that folders that begin with a dot, like `.dbt` are hidden folders in macOS. `profiles.yml` is just a plain text file, so once we find it we can easily edit it. First, open a new Finder window and make sure Finder displays your home directory under "Favorites" on the left side of the window. If it isn't there, select Finder -> Preferences... using the menu bar, select "Sidebar" in the preferences window, and check the box next to the home directory.

After navigating to the home directory in a finder window, use the shortcut **Shift - Command - .** to show hidden folders. You should then see the `.dbt` folder appear and be able to open `profiles.yml` with TextEdit and edit your profile. After you do, use **Shift - Command - .** again to hide the hidden folders.

If you're comfortable with a command-line text editor, like `nano` or `vim`, you can also use that to edit the file via the terminal.
"

Reply:
Personally I feel like the instructions to access via Finder are harder than just explaining to people how to use `vim`, especially since they can copy-paste and only edit one field (name)? But if you think Finder is worth it that's ok with me

Reply:
Honestly not sure, on the one hand yeah it seems simple to use `vim` but writing out those instructions in an equally verbose way also seems complex: 

> First, open your profile using `vim` by running `vim ~/.dbt/profiles.yml`, it should be blank(?).
> Copy the template below, then return to the terminal window with `vim` running. Press i to enter insert mode, then Command-V to paste the template. Press escape to exit insert mode, and move the cursor to the `schema: ` line by using the k and j keys to move up and down, and the h and l keys to move left and right.
> With the cursor on the < character of that line, replace the comment with your name by typing c then $, entering your name, and pressing escape.
> If all looks good, press :wq then enter to save the file.

Like those instructions would probably work, but then if something goes wrong I think a new person would be really stuck. I know the Finder instructions are somewhat convoluted, but I'm hoping they leverage familiar concepts.

Reply:
Yeah fair.... In between could be just having them edit the YML in a text editor and then paste it in using vim?

Reply:
That works for me! Probably more future-proof for when we eventually support Windows/WSL, too

Reply:
I think I am going to include `nano ~/.dbt/profiles.yml` instructions... that prevents the messiness of GUI text editing (and possibly messing up indentation from pasting) but also is easier than using `vim`.

Reply:
That also sounds good